index.js

```diff
@@ -133,6 +139,68 @@ Shopify.prototype.request = function request(url, method, key, params) {
   });
 };
 
+/**
+ * Updates GraphQL API call limits.
+ *
+ * @param {String} throttle The status returned in the GraphQL response
+ * @private
+ */
+Shopify.prototype.updateGqlLimits = function updateGqlLimits(throttle) {
+  const limits = this.callGraphqlLimits;
+
+  limits.remaining = throttle.currentlyAvailable;
+  limits.current = throttle.maximumAvailable - throttle.currentlyAvailable;
+  limits.max = throttle.maximumAvailable;
+
+  this.emit('callGraphqlLimits', limits);
+};
+
+/**
+ * Sends a request to the Shopify GraphQL API endpoint.
+ *
+ * @param string [data] Request body
+ * @return {Promise}
+ * @public
+ */
+Shopify.prototype.graphql = function graphql(data) {
+  const url = assign({ path: '/admin/api/graphql.json' }, this.baseUrl);
+  const options = assign({
+    headers: {
+      'User-Agent': `${pkg.name}/${pkg.version}`,
+      'Content-Type': 'application/graphql'
+    },
+    timeout: this.options.timeout,
+    retries: 0,
+    method: 'POST',
+    body: data
+  }, url);
+
+  if (this.options.accessToken) {
+    options.headers['X-Shopify-Access-Token'] = this.options.accessToken;
+  }
+
+  return got(options).then(res => {
+    try {
+      res.body = JSON.parse(res.body);
+    } catch (err) {
+      const opts = assign({
```

Comment:
This can be simplified  as follows:

```js
const opts = assign({
  host: options.hostname,
  url: urlLib.resolve(urlLib.format(options), options.path)
}, options);
```

Reply:
Right, my bad, sorry

Reply:
Done